### DISCUSSION TITLE
[Chapter 09] 메타프로그래밍 관련 퀴즈

### DISCUSSION THREAD

**Discussion by snoopingh** [cowaine/is-my-code-that-strange#84] [Chapter 09-10 (악마, 이름 설계)]
스프링 프레임워크에서 자주 사용하는 @Controller, @Service, @Repository, @Aspect 같은 어노테이션들은 모두 리플렉션을 활용한 메타프로그래밍의 일종인데요, 그렇다면 롬복 라이브러리에서 지원하는 어노테이션들도 리플렉션을 활용한 메타프로그래밍일까요?

**Comment by firewoody237** [comment]
롬복의 `@Getter`, `@Data` 등 어노테이션 -> `RetentionPolicy.SOURCE` -> 컴파일 시 정의됨

`@Controller`, `@Service` -> `RetentionPolicy.RUNTIME` -> 런타임시 정의 됨

리플렉션은 런타임시 사용되므로, 롬복 라이브러리는 리플렉션을 활용했다고 보기 어려울 것 같습니다.

**Comment by coalong** [comment]
<img width="925" alt="image" src="https://github.com/user-attachments/assets/aa54a9b1-69a5-4cc7-84f5-bc59a9c4fe39">
<img width="925" alt="image" src="https://github.com/user-attachments/assets/0db29b01-0324-423e-88ce-e1bc2a6b0e92">

롬복의 어노테이션은 IntelliJ 디롬복하면 위와 같이 코드를 생성해주네요. @firewoody237 님이 말씀해주신 대로 컴파일시 정의되므로 리플렉션을 활용하진 않은 것 같습니다

**Comment by Irisation23** [comment]
https://free-strings.blogspot.com/2015/12/lombok.html
